selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/WaitUtils.java

```diff
@@ -46,4 +60,79 @@ public static void sleepQuietly(int timeout, TimeUnit timeUnit) {
       throw new RuntimeException(e);
     }
   }
+
+  /**
+   * Waits during {@code timeout} until {@code condition} has a "true" state.
+   *
+   * @param condition expression which should be performed
+   * @param timeout waiting time
+   * @param delayBetweenAttemptsInMilliseconds delay between tries of {@code condition} execution in
+   *     milliseconds
+   */
+  public static void waitSuccessCondition(
+      BooleanSupplier condition,
+      long timeout,
+      long delayBetweenAttemptsInMilliseconds,
+      TimeUnit timeoutTimeUnit)
+      throws InterruptedException {
+    final long waitingTime = timeoutTimeUnit.toMillis(timeout);
+    final long startingTime = System.currentTimeMillis();
+    final long finishTime = startingTime + waitingTime;
+
+    Callable<Boolean> waitTrueState =
+        () -> {
+          while (System.currentTimeMillis() <= finishTime) {
```

Comment:
If `System.currentTimeMillis() = finishTime` it doesn't make sense to continue execution so as timeout exceeded.